api/src/main/java/io/strimzi/api/kafka/model/KafkaUserPlainAuthentication.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@EqualsAndHashCode
+public class KafkaUserPlainAuthentication extends KafkaUserAuthentication {
```

Comment:
This part implements the user management for the User Operator. Do you want to use the User Operator and the `KafkaUser` resources? If yes, we should keep this but we also need to add the implementation for the User Operator. If not we do not need this class. We can also split this into two parts to make it easier to review and merge the PR.

We should probably also follow similar schema as for the Kafka broker configuration - name the class with he Vault, change the type to `sasl-plaintext-vault` (see the comment below).